### DISCUSSION TITLE
Number of cycles of battery deye - sunsynk - three-phase -HV

### DISCUSSION THREAD

**Discussion by JanKraslice** [kellerza/sunsynk#369] [Ideas]
Hello, please is there any possibility to add a sensor to read the battery cycle count? 
I can't find it in the documentation, I would like to see how many cycles my battery has connected to BMS1 and then in the future to BMS2

**Comment by Waz-Cpt** [comment]
I do not this the cycle count is available through the inverter's polling data from the batteries.

Are you batteries connected via CAN or RS485 to the inverter? I would suggest you look for a solution the can connect directly to the BMS while retaining the connection to the inverters.

**Reply by JanKraslice** [reply]
Thank you
I found on 10056 ( Cycle number) within the modbus register to the DEYE inverter it is the battery reading part. 

Yes I use the classic by connecting the inverter with BMS 1 and the other battery BMS 2 
I read the data from the inverter 
The inverter does not display this, but according to DEYE support this information should be available in the registers.

**Comment by JanKraslice** [comment]
Hello, 
line 611 is pack 1 Cycle number this value is also read by solarman
line 625 is pack 2 Cycle number this value is also read by solarman

In this case pack 1 or 2 means battery 1 or 2 see current modbus deye

Can you advise me how to add?
![7 12 2024](https://github.com/user-attachments/assets/f42c7eaf-ca31-4ef0-a98b-3b99ceeb7d78)

**Comment by JanKraslice** [comment]
cycle count value is also available via solarman
or is this value really available via inverter communication.
Attached is a picture of the solarman website

![Snímek obrazovky 2024-12-19 004116](https://github.com/user-attachments/assets/a6a1376d-12d6-483e-9312-3039273766cc)

**Comment by kellerza** [comment]
@JanKraslice have you tried adding a custom sensor to read this?

something like this (needs the imports you can find in the docs)

```python
SENSORS += Sensor(611, "Battery pack 1 cycles", "")
```

**Reply by maslyankov** [reply]
I'd try:

```
SENSORS += Sensor(10049, "Battery pack 1 Remaining Capacity", "AH", 0.1)
SENSORS += Sensor(10050, "Battery pack 1 Total Capacity", "AH", 0.1)
SENSORS += Sensor(10056, "Battery pack 1 Cycles Number", "cycles")
```

**Comment by JanKraslice** [comment]
Hi, thank you.
1) I can't add my own sensor
2) I have tried the one listed in e6a5c0b and it works, it is NOT the number of cycles the battery has completed during its lifetime. 

If you send me exact instructions or pictures I will try to create a custom sensor, but I have never really done it. I am more of a passive user.

I have sent the actual Modbus log to the DEYE HV inverter I have to your email kellerza@.

Thank you very much I appreciate your work and if there is an opportunity to contribute I would be happy to contribute to your work.

**Reply by kellerza** [reply]
Have you tried the docs? <https://kellerza.github.io/sunsynk/reference/mysensors>

**Reply by JanKraslice** [reply]
Yes I have tried before but failed with another sensor.

I have correctly in the file only a line of text, I attach the attachment
SENSORS += Sensor(10056, "Battery pack 1 cycles")

I'm really just a passive user, I can copy or follow the instructions, but I need the exact wording.

Thank you 
[mysensors.py.txt](https://github.com/user-attachments/files/18338131/mysensors.py.txt)

**Reply by maslyankov** [reply]
> Yes I have tried before but failed with another sensor.
> 
> 
> 
> I have correctly in the file only a line of text, I attach the attachment
> 
> SENSORS += Sensor(10056, "Battery pack 1 cycles")
> 
> 
> 
> I'm really just a passive user, I can copy or follow the instructions, but I need the exact wording.
> 
> 
> 
> Thank you 
> 
> [mysensors.py.txt](https://github.com/user-attachments/files/18338131/mysensors.py.txt)
> 
> 

Make sure to remove the ".txt" from the name of the file.

**Reply by kellerza** [reply]
Once you have the correct filename you will see messages, or errors, in the addon's log

With the imports, the full file should be

```python
from sunsynk.sensors import Sensor, SensorDefinitions

SENSORS = SensorDefinitions()

SENSORS += Sensor(10056, "Battery pack 1 cycles")
```
